include/deal.II/matrix_free/mapping_info.h

```diff
@@ -108,7 +108,10 @@ namespace internal
      *
      * @author Katharina Kormann, Martin Kronbichler, 2018
      */
-    template <int structdim, int spacedim, typename Number>
+    template <int structdim,
+              int spacedim,
+              typename Number,
+              typename VectorizedArrayType>
```

Comment:
Here (and in other places), you chose to leave the VectorizedArray type opaque. This means there is no connection between ``Number`` and ``VectorizedArray::Number`` anymore.
You could instead only introduce ``Width`` as a template parameter here and then use ``VectorizedArray<Number, Width>`` below.
Is there a reason for your choice? Is it because Width has a default that you don't want to introduce everywhere?
Would it be possible to insert a ``static_assert`` to make sure ``Number==VectorizedArray::Number`` (the latter doesn't exist yet, sadly)?

Reply:
@peterrum and I discussed this. The idea was that one could, by some additional extensions, mostly related to the array access operator `[]`, also use plain `double` types in `MatrixFree` by the current approach. Now that would be work that neither @peterrum nor I would do at the moment, but it could be done without a break in compatibility.

That said, I would go for a `static_assert` as you suggest, and add `VectorizedArray::value_type` to those classes. (I guess `value_type` is the most appropriate name for it.)

Reply:
I agree (and the name make sense).

Reply:
Hope I did not miss any instances...